includes/admin/class-wc-admin-meta-boxes.php

```diff
@@ -222,6 +224,32 @@ public function save_meta_boxes( $post_id, $post ) {
 			do_action( 'woocommerce_process_' . $post->post_type . '_meta', $post_id, $post );
 		}
 	}
+
+	/**
+	 * Remove block-based templates from the list of available templates for products.
+	 *
+	 * @param string[] $templates Array of template header names keyed by the template file name.
+	 *
+	 * @return string[] Templates array excluding block-based templates.
+	 */
+	public function remove_block_templates( $templates ) {
+		if ( count( $templates ) === 0 ) {
+			return $templates;
+		}
+
+		$theme              = wp_get_theme()->get_stylesheet();
+		$filtered_templates = array();
+
+		foreach ( $templates as $template_key => $template_name ) {
+			$gutenberg_template = gutenberg_get_block_template( $theme . '//' . $template_key );
```

Comment:
Is this function WP version specific? If so, there should probably be a `function_exists` check before calling it (maybe in the initial check at the beginning of the function logic)?